File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/HeWeiSdeLiangGeShuZiLcof.java
package com.dbc.algorithm.ReferOffer;

public class HeWeiSdeLiangGeShuZiLcof {
    public int[] twoSum(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        for (int num : nums) {
            if (nums[left] + nums[right] == target) {
                return new int[]{nums[left], nums[right]};
            } else if (nums[left] + nums[right] < target) {
                left++;
            } else {
                right--;
            }
        }
        return new int[2];
    }
}

File: core/tmp/Python/refer-offer/he_wei_sde_liang_ge_shu_zi_lcof.py
from typing import *

def twoSum(nums: List[int], target: int) -> List[int]:
    left, right = 0, len(nums) - 1
    while left < right:
        if nums[left] + nums[right] == target:
            return [nums[left], nums[right]]
        elif nums[left] + nums[right] < target:
            left += 1
        else:
            right -= 1
    
    return []